### PR TITLE
Adding the possibility to add imagePullSecrets

### PR DESCRIPTION
**Pull request by zonArt**
to the tiller ServiceAccount

### CODE REVIEW

**Comment by edomaur** [comment]
it would be nice, adding imagePullSecrets where they are missing

**Comment by florindragos** [comment]
Hello @zonArt,
Thanks for your contribution!
There seems to be a problem causing the build to fails:

`Syntax error at '|' (file: templates/tiller-serviceaccount.yaml.epp, line: 6, column: 32)`

**Comment by zonArt** [comment]
@florindragos My bad, I'm not super familiar with epp templates. Should be fixed now

**Comment by florindragos** [comment]
No worries! There's a test failing now:
```
Failures:
  1) helm::account_config with default values for all parameters is expected to compile into a catalogue without dependency cycles
     Failure/Error: is_expected.to compile
       error during compilation: Evaluation Error: Unknown variable: 'helm::tiller_image_pull_secrets'. (file: /home/travis/build/puppetlabs/puppetlabs-helm/spec/fixtures/modules/helm/manifests/account_config.pp, line: 6, column: 56) on node travis-job-61328da3-2d9c-402e-ad15-aaea9f6c72b6.c.travis-ci-prod-2.internal
     # ./spec/classes/account_config_spec.rb:14:in `block (3 levels) in <top (required)>'
```

**Comment by zonArt** [comment]
I noticed yes but am not sure to understand what's the issue here, the value exists, doesn't it?

**Comment by florindragos** [comment]
i think it needs to be added to https://github.com/puppetlabs/puppetlabs-helm/blob/master/spec/classes/account_config_spec.rb#L8

**Comment by zonArt** [comment]
Have done it right now but I'm not really convinced this should be need, param is optional so it should compile just fine even without the need to specify it in the test class. Well.... let see what'll happen now that it has been added

**Comment by edomaur** [comment]
Seems to be green :-)

**Comment by zonArt** [comment]
Looks better now, what do you think? Could this be something you envisage for merging? Or do you want me to add further tests on this newly included param?

**Comment by florindragos** [comment]
More tests are always welcome, but it looks good! :-)
Let me just run the acceptance tests.

**Comment by florindragos** [comment]
Looks like helm is not installing, using only `include helm`: https://github.com/puppetlabs/puppetlabs-helm/blob/master/spec/acceptance/helm_spec.rb#L48

```
16:39:31   1) the helm module helm class it should install the module should run
16:39:31      Failure/Error: apply_manifest(pp, :catch_failures => true)
16:39:31      Beaker::Host::CommandFailure:
16:39:31        Host 'cognate-glen.delivery.puppetlabs.net' exited with 1 running:
16:39:31         puppet apply --verbose --detailed-exitcodes /tmp/apply_manifest.pp.XTvFyd
16:39:31        Last 10 lines of output were:
16:39:31        	Info: Loading facts
16:39:31        	Info: Loading facts
16:39:31        	Info: Loading facts
16:39:31        	Info: Loading facts
16:39:31        	Error: Evaluation Error: Error while evaluating a Function Call, lambda: parameter 'image_pull_secrets' expects an Array value, got Undef (file: /etc/puppetlabs/code/modules/helm/manifests/account_config.pp, line: 28, column: 16) on node cognate-glen.delivery.puppetlabs.net
16:39:31        
16:39:31      # ./.bundle/gems/gems/beaker-4.14.0/lib/beaker/host.rb:391:in `exec'
16:39:31      # ./.bundle/gems/gems/beaker-4.14.0/lib/beaker/dsl/helpers/host_helpers.rb:83:in `block in on'
16:39:31      # ./.bundle/gems/gems/beaker-4.14.0/lib/beaker/shared/host_manager.rb:130:in `run_block_on'
16:39:31      # ./.bundle/gems/gems/beaker-4.14.0/lib/beaker/dsl/patterns.rb:37:in `block_on'
16:39:31      # ./.bundle/gems/gems/beaker-4.14.0/lib/beaker/dsl/helpers/host_helpers.rb:63:in `on'
16:39:31      # ./.bundle/gems/gems/beaker-puppet-1.18.8/lib/beaker-puppet/helpers/puppet_helpers.rb:529:in `block in apply_manifest_on'
16:39:31      # ./.bundle/gems/gems/beaker-4.14.0/lib/beaker/shared/host_manager.rb:130:in `run_block_on'
16:39:31      # ./.bundle/gems/gems/beaker-4.14.0/lib/beaker/dsl/patterns.rb:37:in `block_on'
16:39:31      # ./.bundle/gems/gems/beaker-puppet-1.18.8/lib/beaker-puppet/helpers/puppet_helpers.rb:457:in `apply_manifest_on'
16:39:31      # ./.bundle/gems/gems/beaker-puppet-1.18.8/lib/beaker-puppet/helpers/puppet_helpers.rb:536:in `apply_manifest'
16:39:31      # ./spec/acceptance/helm_spec.rb:48:in `block (4 levels) in <top (required)>'
```

File: manifests/account_config.pp [hunk 1]
```diff
@@ -25,7 +26,8 @@
     group   => 'root',
     mode    => '0644',
     content => epp('helm/tiller-serviceaccount.yaml.epp', {
-      'service_account' => $service_account,
+      'service_account'    => $service_account,
+      'image_pull_secrets' => $tiller_image_pull_secrets,
```
Comment (florindragos) [inline manifests/account_config.pp]: add a check to only add this line unless `$tiller_image_pull_secrets == undef` ?

Comment (florindragos) [reply manifests/account_config.pp]: If the configuration accepts an empty array, setting the default in params.pp to `[]` would probably also be an option.

Comment (zonArt) [reply manifests/account_config.pp]: You're fully right, sorry for it, will do

**Comment by zonArt** [comment]
[] is accepted in SA's configuration so I changed the default as suggested, thanks

**Comment by florindragos** [comment]
```
10:59:06 Finished in 6 minutes 9 seconds (files took 15.2 seconds to load)
10:59:06 17 examples, 0 failures
```

**Pull request merged**